packages/ui-react/src/components/Sidebar/Sidebar.tsx

```diff
@@ -55,6 +52,30 @@ const Sidebar: React.FC<SidebarProps> = ({ isOpen, onClose, children }) => {
     };
   }, [isOpen, onClose]);
 
+  useEffect(() => {
+    const sidebarElement = sidebarRef.current;
+    const handleTransitionEnd = () => {
+      setVisible(isOpen);
+    };
+
+    sidebarElement?.addEventListener('transitionend', handleTransitionEnd);
+
+    return () => {
+      sidebarElement?.removeEventListener('transitionend', handleTransitionEnd);
+    };
+  }, [isOpen]);
+
+  useEffect(() => {
+    lastFocusedElementRef.current = document.activeElement as HTMLElement;
+
+    if (visible) {
+      sidebarRef.current?.querySelectorAll('a')[0]?.focus({ preventScroll: true });
+    } else {
+      lastFocusedElementRef.current?.focus({ preventScroll: true });
+      setVisible(false);
```

Comment:
This `setVisible(false);` not needed (it is double). The first `useEffect` handles this state already.